ArmPkg/Library/TpmServiceLib/TpmServiceLib.c

```diff
@@ -0,0 +1,333 @@
+/** @file
+  Implementation for the TPM Service
+
+  Copyright (c), Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/TpmServiceLib.h>
+#include <Library/Tpm2DeviceLib.h>
+#include <IndustryStandard/TpmPtp.h>
+
+/* TPM Service Defines */
+#define TPM_MAJOR_VER              (1)
+#define TPM_MINOR_VER              (0)
+
+#define TPM_START_PROCESS_CMD      (0)
+#define TPM_START_PROCESS_LOC_REQ  (1)
+
+#define TPM_LOC_STATE_MASK         (0x9F)
+#define TPM_LOC_STS_MASK           (0x01)
+#define TPM_CTRL_START_MASK        (0x01)

```

Comment:
nit: do you think it makes sense to have these live in the lib header?

Reply:
No because they aren't needed outside of the .c file. These are only needed by the cleanup function which makes sure the CRB is in a good state.